### PR TITLE
CH-1081 Migrate and remove v1 purchase API documentation

### PR DESCRIPTION
**Pull request by martinfridh**
- [x] First go at setting structure for updated purchase API documentation
- [x] Remove reference to swagger doc (contains too much stuff, including v1 endpoints) not suitable for external integrators 
- [x] Removing public mention of that WRITE_PURCHASE is available scope for purchase service
- [x] Add 3-year age limit of purchases returned, document it, and remove info on legacy uuid format
- [x] Add more payments info
- [x] Update payments data in purchase examples
- [x] Add info on max limit to be used (i.e. 100)

### CODE REVIEW

File: purchase.adoc [hunk 1]
```diff
@@ -443,6 +429,96 @@ purchaseUUID:: The UUID of the purchase
...
+    {
+      "type": "IZETTLE_INVOICE",
+      ...
+    },
+  ],
+  ...
+}
+----
+
+In addition to `IZETTLE_CARD`, `IZETTLE_CARD_ONLINE`, `IZETTLE_CASH` and `IZETTLE_INVOICE`, there is also payment types of `SWISH`, `VIPPS`, `MOBILE_PAY`, `PAYPAL`, `STORE_CREDIT` and `CUSTOM`.
```
Comment (izjorgen) [inline purchase.adoc]: How about GIFTCARD, KLARNA etc?  How much of the future payment should we expose here?

File: purchase.adoc [hunk 2]
```diff
@@ -385,7 +371,7 @@ purchaseUUID:: The UUID of the purchase
 404:: Purchase not found
 
 #### Example response
-`GET /purchase/6HbDrnUNRji5iniGikNLiQ`
+`GET /purchase/v2/6HbDrnUNRji5iniGikNLiQ`
```
[the remark this answers is not in the document]
Comment (martinfridh) [reply purchase.adoc]: Good idea!

File: purchase.adoc [hunk 3]
```diff
@@ -443,6 +429,96 @@ purchaseUUID:: The UUID of the purchase
 }
 ```
 
+### Payments
+
+TODO: Add some more overview info
+
+._example: Version 2 purchase with one card, one cash and one invoice payment_
```
[the remark this answers is not in the document]
Comment (martinfridh) [reply purchase.adoc]: Well, the payments model at least is based on payments in list so I guess that is why we explain it like this.

File: purchase.adoc (hunk 1 above)
Comment (martinfridh) [reply purchase.adoc]: Good point! My spontaneous thought is that we should probably be cautious about sharing info on not yet released products, so I would suggest adding the payment types when more officially launched.

File: purchase.adoc [hunk 4]
```diff
@@ -11,45 +11,30 @@ The Purchase API implements the following scopes:
...
-link:purchase_v2.adoc[*Version 2 documentation*]
-
-._OBSERVE!_
-____
-Due to backwards compatability requirements with some of our clients, some endpoints in the *_version 1_* API will not return purchases that lack `cardPayments` or `cashPayments`. This means that any purchase that was paid by invoice – or by some payment type other than card or cash – may be missing in the results. To be sure to receive all purchases you must use the version 2 API.
-____
-
 === Get purchase history
-Will include cash register information if possible.
+Will include cash register information if possible. Purchases up to three years old are available to fetch.
```
[the remark this answers is not in the document]
Comment (martinfridh) [reply purchase.adoc]: I guess "if applicable" would be more suitable.

File: purchase.adoc [hunk 5]
```diff
@@ -11,45 +11,30 @@ The Purchase API implements the following scopes:
...
-`GET /purchases`
+`GET /purchases/v2`
 
 lastPurchaseHash (optional):: A value from "lastPurchaseHash" from the result of a previous history query, to continue listing purchases from the next record after the previous query.
-startDate (optional):: The start date (inclusive) for purchases to be retrieved from until today or endDate. Can only be combined with endDate or limit.
-endDate (optional):: The last date (exclusive) for purchases to be retrieved until. Can only be combined with startDate.
-limit (optional):: The maximum number of records to return.
+startDate (optional):: The start date (inclusive) for purchases to be retrieved from until today or endDate. By default startDate is resolved to three years back.
+endDate (optional):: The last date (exclusive) for purchases to be retrieved until.
+limit (optional):: The maximum number of records to return. Max value of limit is 1000.
```
[the remark this answers is not in the document]
Comment (martinfridh) [reply purchase.adoc]: Actually, both max and default value is currently `100 000`. Reason for here stating it to be `1 000` is to prepare integrators for us to change the default/max value at later point, hopefully during autumn of 2019...

File: purchase.adoc [hunk 6]
```diff
@@ -1,4 +1,4 @@
-## Purchase API V1
+## Purchase API
 
 ### Service Description
 The purchase API provides information about purchases made through the iZettle point-of-sale.
```
[the remark this answers is not in the document]
Comment (martinfridh) [reply purchase.adoc]: Good point! 👍

**Pull request merged**